### ISSUE TITLE
⚠️ Do not expose access key and hostname in logs

### ISSUE THREAD

**Issue by timmo001** [Customatics/VSTS-TFS-Integration#1]
Anyone that has access to the pipeline logs in Azure DevOps will be able to see this sensitive data. This is fine for debugging purposes, but these are very sensitive details that shouldn't be exposed.

Here are the job logs where this is occuring:

```
2019-08-05 09:54:54.952 C:\apps\exe\vsts-agent-win-x64-2.150.3\_work\1\s\LeapworkIntegrationAgent11.Build51.log is created.
2019-08-05 09:54:54.955 [INFO]  LEAPWORK Plugin input parameters:  
2019-08-05 09:54:54.955 [INFO]  ----------------------------------------------------------------------------------------  
2019-08-05 09:54:54.955 [INFO]  LEAPWORK controller hostname: MYSECRETHOSTNAME  
2019-08-05 09:54:54.956 [INFO]  LEAPWORK controller port: 9001  
2019-08-05 09:54:54.956 [INFO]  LEAPWORK controller Access Key: MYSUPERSECRETKEY
2019-08-05 09:54:54.956 [INFO]  JUnit report file name: C:\apps\exe\vsts-agent-win-x64-2.150.3\_work\1\s\repo
```

**Comment by timmo001** [comment]
An odd thing though, is the logs are printed twice. Further down, where the test has finished, the `ps1` log prints out, but with the relavant sensitive info obfuscated:

```
##[section]Starting: LeapworkIntegration
==============================================================================
Task         : Leapwork Integration
Description  : Run Leapwork schedules
Version      : 3.1.3
Author       : Customatics
Help         : [More Information](https://leapwork.com/)
==============================================================================
##[warning]Task 'Leapwork Integration' (3.1.3) is using deprecated task execution handler. The task should use the supported task-lib: https://aka.ms/tasklib
Preparing task execution handler.
Executing the powershell script: C:\apps\exe\vsts-agent-win-x64-2.150.3\_work\_tasks\Leapwork Integration_4c073640-a475-11e5-a7e5-850fa74c7a56\3.1.3\Script.ps1
LEAPWORK Plugin input parameters:
----------------------------------------------------------------------------------------
LEAPWORK controller hostname: ***
LEAPWORK controller port: 9001
LEAPWORK controller Access Key: ***
JUnit report file name: C:\apps\exe\vsts-agent-win-x64-2.150.3\_work\1\s\report.xml
Log file path: C:\apps\exe\vsts-agent-win-x64-2.150.3\_work\1\s\LeapworkIntegrationAgent11.Build52.log
```

**Comment by romanchetto** [comment]
Hello @timmo001 , Thank you for your advice.
Just published new version 4.0.0 run by powershell 3 and with hidden Access Key. It was decided not to hide hostname as incorrect hostname is very widespread issue and it helps to detect it fast.
